packages/ui/ToggleItem/ToggleItem.js

```diff
@@ -24,17 +24,42 @@ import Touchable from '../Touchable';
 
 import componentCss from './ToggleItem.module.less';
 
-// eslint-disable-next-line
-const iconCreator = (position) => ({disabled, icon, iconComponent, selected, iconPosition}) => {
-	if (iconPosition === position) {
+// eslint-disable-next-line enact/display-name,enact/prop-types
+const iconCreator = (position) => ({disabled, icon, iconComponent, iconPosition, itemIcon, itemIconPosition, selected}) => {
+
+	const itemIconComponent = (itemIcon ? <ComponentOverride component={itemIcon} /> : null);
```

Comment:
I'm not sure this is required but wanted to check on the use cases. The docs show (and the samples above demonstrate) accepting a React Element (`<Icon>star</Icon>`). `ComponentOverride` accepts either a component (e.g. `Icon`) or an element so it "works" here but it's not passing any props down so it's only value is taking a component (which the docs don't say we support) and returning an element. However, without passing any props along, the icon/content would have to be built into the component which is the thing we wanted to avoid requiring by introducing `ComponentOverride` in the first place!

That might not make sense -- it's a somewhat complex arrangement -- but it simplifies down to:
* Do we need to support components as a value for `itemIcon`?

If not, then the above code can be reduce to just using `itemIcon` and we can skip `ComponentOverride` altogether. That would be my inclination given what I see right now.

Reply:
@ryanjduffy ,
What did you mean `components as a value for itemIcon`? Did you ask if other components besides the Icon component could be used?
Up to now, all use cases only use icon. (Actually this feature has customized with `Image` by applications. However, once we support this component, it's possible to satisfy that requirements with `Icon`.)

Reply:
I'm trying to distinguish between Components (the constructor of elements) and elements (the instances of a component that are rendered). I want to understand if there's a need to support both or just elements (as the docs currently state).

Reply:
I've removed the `ComponentOverride` usage here based on my above observations. If we later decide it would be valuable to support the use case of a Component rather than an element here, we can add it back and update the docs to reflect that.